### DISCUSSION TITLE
$site.pages 怎么迁移

### DISCUSSION THREAD

**Discussion by zhouxinyong** [vuepress/core#549] [Q&A] [answered]
我在迁移theme 到V2版本时，usePagesData的返回值也并不和$site.pages一样。

文档里也没有具体的迁移说明。原来的$site.pages该用什么方式拿到呢

**Comment by Mister-Hope** [comment, marked as answer]
因为可扩展性的考虑移除了，你不应该试图在客户端侧获取全部的页面对象。

另外你可以使用 routeMeta 去向路由注入一些必要的页面元信息。

你可以关注一下 vuepress-theme-hope，想实现的东西里面都应该有类似地实现。

**Reply by zhouxinyong** [reply]
这块没有可替代方案吗？

**Reply by Mister-Hope** [reply]
现在有了，你可以使用 routeMeta。

你可以关注一下 vuepress-theme-hope，你想实现的东西里面都应该有类似地实现。

**Comment by hustcer** [comment]
由于特殊需要我们确实需要得到所有页面的路径，有什么办法吗？急，谢谢，之前是用：
```vue
<script>
  export default {
    computed: {
      commands() {
        return this.$site.pages
          .filter(p => p.path.indexOf('/book/commands/') >= 0)
          .sort((a,b) => (a.title > b.title) ? 1 : ((b.title > a.title) ? -1 : 0));
      }
    }
  }
</script>
```

**Reply by meteorlxy** [reply]
```js
// .vuepress/config.js
//...
  onPrepared: async (app) => {
    await app.writeTemp(
      'pages.js',
      `export default ${JSON.stringify(app.pages.map(({ data }) => data))}`
    )
  },
//...
```

```js
import pages from '@temp/pages'
```

**Reply by meteorlxy** [reply]
https://v2.vuepress.vuejs.org/advanced/cookbook/passing-data-to-client-code.html#write-and-load-temp-files

**Reply by hustcer** [reply]
@meteorlxy That's great, thanks very much. 👍 👍 👍 👍 👍
